aws/s3_test.go

```diff
@@ -0,0 +1,148 @@
+package aws
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/cloud-nuke/util"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func createTestS3Bucket(t *testing.T, session *session.Session, bucketName string) {
```

Comment:
This is a good start for testing, but I'd like to see:

- A bucket with objects to make sure cloud-nuke handles that correctly
- A bucket with versioning enabled, and a deleted object version to make sure that is handled.